Failures with Python 3.11

See https://docs.python.org/3.10/library/asyncio-task.html#generator-based-coroutines

```bash
============================= test session starts ==============================
platform linux -- Python 3.11.1, pytest-7.2.0, pluggy-1.0.0
rootdir: /build/source
plugins: aresponses-2.1.6, asyncio-0.20.3
asyncio: mode=Mode.STRICT
collected 29 items / 1 error                                                   

==================================== ERRORS ====================================
_________________ ERROR collecting tests/test_feed_manager.py __________________
tests/test_feed_manager.py:4: in <module>
    from asynctest import CoroutineMock, patch
/nix/store/701b4291rivd2q69l2rxxc74gcasbp1p-python3.11-asynctest-0.13.0/lib/python3.11/site-packages/asynctest/__init__.py:22: in <module>
    from .case import *
/nix/store/701b4291rivd2q69l2rxxc74gcasbp1p-python3.11-asynctest-0.13.0/lib/python3.11/site-packages/asynctest/case.py:54: in <module>
    import asynctest.selector
/nix/store/701b4291rivd2q69l2rxxc74gcasbp1p-python3.11-asynctest-0.13.0/lib/python3.11/site-packages/asynctest/selector.py:29: in <module>
    from . import mock
/nix/store/701b4291rivd2q69l2rxxc74gcasbp1p-python3.11-asynctest-0.13.0/lib/python3.11/site-packages/asynctest/mock.py:428: in <module>
    class _AwaitEvent:
/nix/store/701b4291rivd2q69l2rxxc74gcasbp1p-python3.11-asynctest-0.13.0/lib/python3.11/site-packages/asynctest/mock.py:433: in _AwaitEvent
    @asyncio.coroutine
E   AttributeError: module 'asyncio' has no attribute 'coroutine'
=========================== short test summary info ============================
ERROR tests/test_feed_manager.py - AttributeError: module 'asyncio' has no attribute 'coroutine'
```

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -1,5 +1,5 @@
 [tool.black]
-target-version = ['py37', 'py38', 'py39', 'py310']
+target-version = ['py37', 'py38', 'py39', 'py310', 'py311']
 
 [tool.isort]
 profile = "black"

--- a/setup.py
+++ b/setup.py
@@ -39,6 +39,7 @@ setup(
         "Programming Language :: Python :: 3.8",
         "Programming Language :: Python :: 3.9",
         "Programming Language :: Python :: 3.10",
+        "Programming Language :: Python :: 3.11",
         "License :: OSI Approved :: Apache Software License",
         "Operating System :: OS Independent",
     ],

--- a/tests/test_feed_manager.py
+++ b/tests/test_feed_manager.py
@@ -1,7 +1,7 @@
 """Test for the generic geojson feed manager."""
 import aiohttp
 import pytest
-from asynctest import CoroutineMock, patch
+import mock as async_mock
 
 from aio_geojson_client.consts import UPDATE_OK_NO_DATA
 from aio_geojson_client.feed_manager import FeedManagerBase
@@ -120,8 +120,8 @@ async def test_feed_manager(aresponses, event_loop):
         updated_entity_external_ids.clear()
         removed_entity_external_ids.clear()
 
-        with patch(
-            "aio_geojson_client.feed.GeoJsonFeed._fetch", new_callable=CoroutineMock
+        with async_mock.patch(
+            "aio_geojson_client.feed.GeoJsonFeed._fetch", new_callable=async_mock.AsyncMock
         ) as mock_fetch:
             mock_fetch.return_value = (UPDATE_OK_NO_DATA, None)
 

--- a/tox.ini
+++ b/tox.ini
@@ -1,5 +1,5 @@
 [tox]
-envlist = py37, py38, py39, py310, cov, cov_local
+envlist = py37, py38, py39, py310, py311, cov, cov_local
 skip_missing_interpreters = True
 ignore_basepython_conflict = True
 
@@ -9,12 +9,12 @@ python =
     3.8: cov
     3.9: py39
     3.10: py310
+    3.11: py311
 
 [testenv]
 deps=
     pytest
     mock
-    asynctest
     aresponses
 commands=pytest
 
@@ -24,7 +24,6 @@ deps=
     pytest
     pytest-cov
     mock
-    asynctest
     aresponses
 commands=
     pytest --cov --cov-report= {posargs}
@@ -35,7 +34,6 @@ deps=
     pytest
     pytest-cov
     mock
-    asynctest
     aresponses
 commands=
     pytest --cov --cov-report=

